test/common/event/dispatcher_impl_test.cc

```diff
@@ -472,41 +472,67 @@ TEST_F(DispatcherMonotonicTimeTest, ApproximateMonotonicTime) {
   dispatcher_->run(Dispatcher::RunType::Block);
 }
 
-TEST(TimerImplTest, TimerEnabledDisabled) {
-  Api::ApiPtr api = Api::createApiForTest();
-  DispatcherPtr dispatcher(api->allocateDispatcher("test_thread"));
-  Event::TimerPtr timer = dispatcher->createTimer([] {});
+class TimerImplTest : public testing::Test {
+protected:
+  void SetUp() override {
+    // Update time cache to provide a stable time reference for timer registration.
+    event_base_update_cache_time(&libevent_base_);
+  }
+
+  // Advance time forward while updating the libevent's time cache and monotonic time reference.
+  // Pushing the monotonic time reference forward eliminates the possibility of time moving
+  // backwards and breaking the overly picky TimerImpl tests below.
+  void AdvanceLibeventTime(absl::Duration duration) {
+    timeval start_tv;
+    {
+      int ret = event_base_gettimeofday_cached(&libevent_base_, &start_tv);
+      RELEASE_ASSERT(ret == 0, "event_base_gettimeofday_cached failed");
+    }
+
+    timeval now_tv;
+    do {
+      absl::SleepFor(duration);
```

Comment:
Theoretically we can sleep less in the second loop... 

nvm in test case we sleep 5ms-ish